### ISSUE TITLE
LSP Server not working in Visual Studio

### ISSUE THREAD

**Issue by gep13** [OmniSharp/csharp-language-server-protocol#132]
I have been setting up an LSP implementation for Chocolatey nuspec files in VSCode, and this has been working well.  I am now trying to take the same Language Server implementation and run it in Visual Studio.  When I try this, I get the following error returned:

![image](https://user-images.githubusercontent.com/1271146/54780784-e1cef780-4bd7-11e9-8c34-1734194246a1.png)

The code that I am using the initialise the client within Visual Studio is the following:

```
    [ContentType("nuspec")]
    [Export(typeof(ILanguageClient))]
    public class ChocolateyLanguageClient : ILanguageClient
    {
        public string Name => "Chocolatey Language Server Client";

        public IEnumerable<string> ConfigurationSections => null;

        public object InitializationOptions => null;

        public IEnumerable<string> FilesToWatch => null;

        public event AsyncEventHandler<EventArgs> StartAsync;
        public event AsyncEventHandler<EventArgs> StopAsync;

        public async Task<Connection> ActivateAsync(CancellationToken token)
        {
            await Task.Yield();

            var info = new ProcessStartInfo();
            info.FileName = "dotnet";
            info.Arguments = "C:\\github_local\\gep13\\chocolatey-vs\\lib\\ChocolateyLanguageServer\\Chocolatey.Language.Server.dll";
            info.RedirectStandardInput = true;
            info.RedirectStandardOutput = true;
            info.UseShellExecute = false;
            info.CreateNoWindow = true;

            Process process = new Process();
            process.StartInfo = info;

            if (process.Start())
            {
                return new Connection(process.StandardOutput.BaseStream, process.StandardInput.BaseStream);
            }

            return null;
        }

        public async Task OnLoadedAsync()
        {
            await StartAsync?.InvokeAsync(this, EventArgs.Empty);
        }

        public Task OnServerInitializedAsync()
        {
            return Task.CompletedTask;
        }

        public Task OnServerInitializeFailedAsync(Exception e)
        {
            return Task.CompletedTask;
        }
    }
```

Is there additional configuration that needs to be provided, or is this a bug that needs to be addressed.  Let me know if you need any further information.

**Comment by gep13** [comment]
@mholo65 @david-driscoll @tintoy any ideas about the above?

**Comment by gep13** [comment]
Seems the related code is here:

https://github.com/OmniSharp/csharp-language-server-protocol/blob/9fa60ffdaf7762ef29f8badea6182f272d0e5979/src/Server/ClientCapabilityProvider.cs#L28-L30

I am using the 0.12.0 version of OmniSharp.Extensions.LanguageServer.

**Comment by gep13** [comment]
I had a concern that the issue was the custom implementation of the ConfigurationHandler that I was using, but I have just removed that, and I am still getting the same error.

**Comment by NickAcPT** [comment]
I've had some problems like that when I tried to run my language server on Sublime (with the LSP plugin) and on IntelliJ. Sadly, I no longer have the logs.
Don't quote me on this, but I think it might be something related to static registration since the error doesn't happen for me on vscode.

**Comment by mholo65** [comment]
IIRC, I was able to reproduce using the Client bits in the OmniSharp.Extensions LSP library. I’ll have a look in the upcoming days.

**Comment by mholo65** [comment]
Did some debugging. It currently fails due to the fact that [`SynchronizationCapability`](https://github.com/OmniSharp/csharp-language-server-protocol/blob/7d8567c09c88d27b4b4a2209959f689df8e9419d/src/Protocol/Client/Capabilities/SynchronizationCapability.cs) implements generic interface `ConnectedCapability<T>` more than once. I am able to reproduce via unit tests by adding `typeof(SynchronizationCapability)` to [ClientCapabilityProviderTests](https://github.com/OmniSharp/csharp-language-server-protocol/blob/7d8567c09c88d27b4b4a2209959f689df8e9419d/test/Lsp.Tests/ClientCapabilityProviderTests.cs#L37).

Will try to provide a fix in the upcoming days.

**Comment by gep13** [comment]
@mholo65 nice one!

Any thoughts on why this works in VSCode, but not in Visual Studio?

**Comment by mholo65** [comment]
@gep13 I'd go with same theory as @NickAcPT regarding static and dynamic registrations.

**Comment by gep13** [comment]
@mholo65 ah, sorry, missed that part!

**Comment by mholo65** [comment]
I managed to find a fix for this and submitted a PR #133. Tested it out with OmniSharp, and wasn't able to reproduce the bug anymore. Also tested OmniSharp with VSCode, and everything seemed to work fine.
